parsers/MY_WM.py

```diff
@@ -108,59 +127,49 @@ def fetch_exchange(
-            exchanges[exchange["Tarikhmasa"]] += exchange["MW"]
-        return [
-            {
-                "datetime": arrow.get(timestamp, tzinfo=TIMEZONE).datetime,
-                "netFlow": power,
-                "sortedZoneKeys": sorted_zone_keys,
-                "source": DOMAIN,
-            }
-            for timestamp, power in exchanges.items()
-        ]
+            all_exchange_data.append(
+                datetime=arrow.get(exchange["Tarikhmasa"], tzinfo=TIMEZONE).datetime,
+                netFlow=exchange["MW"],
+                zoneKey=sorted_zone_keys,
+                source=DOMAIN,
+            )
     else:
         raise NotImplementedError("'{sorted_zone_keys}' is not implemented")
+    return all_exchange_data.to_list()
 
 
 @config.refetch_frequency(timedelta(minutes=10))
 def fetch_production(
-    zone_key: str = DEFAULT_ZONE_KEY,
-    session: Optional[Session] = None,
+    zone_key: ZoneKey = DEFAULT_ZONE_KEY,
+    session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> list:
     """Request the production mix (in MW) of a given zone."""
     date_string = arrow.get(target_datetime).to(TIMEZONE).format("DD/MM/YYYY")
-    return [
-        validation.validate(
-            {
-                "datetime": arrow.get(breakdown["DT"], tzinfo=TIMEZONE).datetime,
-                "production": {
-                    "coal": breakdown["Coal"],
-                    "gas": breakdown["Gas"],
-                    "hydro": breakdown["Hydro"],
-                    "oil": breakdown["Oil"],
-                    "solar": breakdown["Solar"],
-                    "unknown": breakdown["CoGen"],
-                },
-                "source": DOMAIN,
-                "zoneKey": zone_key,
-            },
-            logger,
-            floor=PRODUCTION_THRESHOLD,
-            remove_negative=True,
-        )
-        for breakdown in get_api_data(
-            session or Session(),
-            PRODUCTION_URL,
-            {
-                "Fromdate": date_string,
-                "Todate": date_string,
-            },
+    all_production_data = ProductionBreakdownList(logger)
+
+    production_data = get_api_data(
+        session,
+        PRODUCTION_URL,
+        {
+            "Fromdate": date_string,
+            "Todate": date_string,
+        },
+    )
+    for item in production_data:
+        production_mix = ProductionMix()
+        item_datetime = arrow.get(item["DT"], tzinfo=TIMEZONE).datetime
+        for mode in [key for key in item if key != "DT"]:
+            production_mix.add_value(PRODUCTION_BREAKDOWN[mode], item[mode], True)
+        all_production_data.append(
+            zoneKey=zone_key,
+            production=production_mix,
+            source=DOMAIN,
+            datetime=item_datetime,
         )
-    ]
+    return all_production_data.to_list()
 
 
 @config.refetch_frequency(timedelta(minutes=10))
```

Comment:
Do you want to upgrade this function in the parser as well?
Since we are changing all the others I think it makes sense.

Reply:
I didn't update this one because we don't make any use of production forecasts. But for consistency I will